backend/controllers/userControllers.js

```diff
@@ -1,57 +1,72 @@
 const connection = require("../db/connection");
 const bcrypt = require("bcrypt");
 
-const register = async (request, response) => {
-  const { username, password, email } = request.body;
+const register = async (req, res) => {
+  const { username, password, email } = req.body;
   if (username && password && email) {
     const salt = "batteryAnt"; //TODO find out what this is for...
     const imageUrl = "/"; //TODO: change this to the default image url
-    //TODO: check if this username is availabe in the db
-    let selectUserQuery =
-      "SELECT * FROM users WHERE username = $1 OR email = $2";
+    const selectUserQuery =
+      "SELECT * FROM users WHERE username = $1 or email = $2";
     try {
       const result = await connection.query(selectUserQuery, [username, email]);
+      const errors = [];
       if (result.length > 0) {
-        console.log(
-          `username: ${username} is already taken or email ${email} is already in use`,
-        );
-        response.send({ error: "username or email is already taken" });
-        return;
+        result.forEach((user) => {
+          if (user.username === username) {
+            errors.push("username is taken");
+          }
+
+          if (user.email === email) {
+            errors.push("email is taken");
+          }
+        });
+
+        req.session.errors = errors.join("\n");
+        return res.redirect("/register");
       }
     } catch (error) {
-      console.log("error checking if username is available: " + error);
+      console.error(error);
+
+      req.session.errors = "Internal Server Errors";
+      return res.redirect("/register");
     }
 
     bcrypt.hash(password, 10, async (err, hash) => {
       if (err) {
-        console.log("error hashing password: " + err);
-        response.redirect("/registerError");
+        console.error("error hashing password: " + err);
+
+        req.session.errors = "Internal Server Errors";
+        return res.redirect("/register");
```

Comment:
Addressed change requested by @CaiminRO , Register and Login Errors are now stored and passed to the frontend using sessions then redirect back to the register page